Userland/Libraries/LibGUI/RangeSlider.cpp

```diff
@@ -0,0 +1,180 @@
+
+    // Grid pattern
+    Gfx::StylePainter::paint_transparency_grid(painter, inner_rect, palette());
+
+    // Alpha gradient
+    painter.fill_rect_with_linear_gradient(inner_rect, m_background_gradient, orientation() == Orientation::Horizontal ? 90.0f : 180.0f);
+
+    Gfx::StylePainter::paint_button(painter, knob_rect_for_value(lower_range()), palette(), Gfx::ButtonStyle::Normal, false, m_hovered_lower_knob);
+    Gfx::StylePainter::paint_button(painter, knob_rect_for_value(upper_range()), palette(), Gfx::ButtonStyle::Normal, false, m_hovered_upper_knob);
+
+    // Text label
+    if (m_show_label) {
+        auto range_text = DeprecatedString::formatted("{} to {}", lower_range(), upper_range());
+        painter.draw_text(inner_rect.translated(1, 1), range_text, Gfx::TextAlignment::Center, Color::Black);
+        painter.draw_text(inner_rect, range_text, Gfx::TextAlignment::Center, Color::White);
+    }
+
+    // Frame
+    Gfx::StylePainter::paint_frame(painter, rect(), palette(), Gfx::FrameStyle::SunkenContainer);
+}
+
+int RangeSlider::value_at(Gfx::IntPoint position) const
+{
+    auto inner_rect = frame_inner_rect();
+    auto relevant_position = position.primary_offset_for_orientation(orientation()),
+         begin_position = inner_rect.first_edge_for_orientation(orientation()),
+         end_position = inner_rect.last_edge_for_orientation(orientation());
+    if (relevant_position < begin_position)
+        return min();
+    if (relevant_position > end_position)
+        return max();
+    float relative_offset = (float)(relevant_position - begin_position) / (float)inner_rect.primary_size_for_orientation(orientation());
+
+    int range = max() - min();
+    return min() + (int)(relative_offset * (float)range);
+}
+
+void RangeSlider::set_gradient_color(Gfx::Color from_color, Gfx::Color to_color)
+{
+    m_background_gradient = Vector { Gfx::ColorStop { from_color, 0 }, Gfx::ColorStop { to_color, 1 } };
+    update();
+}
+
+void RangeSlider::set_gradient_colors(Vector<Gfx::ColorStop> colors)
+{
+    VERIFY(colors.size());
+    m_background_gradient = colors;
+    update();
+}
+
+void RangeSlider::mousedown_event(MouseEvent& event)
+{
+    if (event.button() == MouseButton::Primary) {
+        m_dragging = true;
+        int clicked_value = value_at(event.position());
+        if (m_hovered_lower_knob)
+            set_lower_range(clicked_value);
+        if (m_hovered_upper_knob)
+            set_upper_range(clicked_value);
+
+        if (!m_hovered_lower_knob && !m_hovered_upper_knob) {
+            if (clicked_value < lower_range())
+                set_lower_range(lower_range() - page_step());
+            if (clicked_value > upper_range())
+                set_upper_range(upper_range() + page_step());
+            if (clicked_value > lower_range() && clicked_value < upper_range()) {
+                set_lower_range(lower_range() + page_step());
+                set_upper_range(upper_range() - page_step());
+            }
```

Comment:
If you click slightly right of the upper range knob, it behaves as if you've clicked in between them.